android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt

```diff
@@ -0,0 +1,53 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.ui.theme.ButtonBlue
+import agency.five.codebase.android.movieapp.ui.theme.spacing
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+
+
+@Composable
+fun FavoriteButton(
+    modifier: Modifier = Modifier,
+    isFavorite: Boolean,
+    onClick: () -> Unit
+) {
+    Image(
+        painter = painterResource(id = if (isFavorite) R.drawable.liked else R.drawable.not_liked),
+        contentDescription = "Liked icon",
+        modifier = modifier
+            .clickable {
+                onClick()
+            }
+            .background(
+                color = ButtonBlue.copy(alpha = 0.6f),
+                shape = CircleShape,
+            )
+            .padding(MaterialTheme.spacing.small),
+    )
+}
+
+fun addToFavorites() {
+    //TODO
+}
```

Comment:
This is unnecessary. 
The whole idea behind state hoisting is to define the behaviour in the highest component which can mutate the state. This logic will be defined later and it will be done somewhere else, right now it is important that you created component which can take onClick logic as argument.